sdk/storage/src/core/clients/storage_client.rs

```diff
@@ -66,6 +66,114 @@ pub enum ServiceType {
     Table,
 }
 
+#[derive(Clone, Debug)]
+pub struct StorageClientBuilder {
+    storage_credentials: Option<StorageCredentials>,
+    blob_storage_url: Url,
+    table_storage_url: Url,
+    queue_storage_url: Url,
+    queue_storage_secondary_url: Url,
+    filesystem_url: Url,
+    account: String,
+    storage_options: StorageOptions,
+}
+
+impl StorageClientBuilder {
+    #[must_use]
+    pub fn new(account: impl Into<String>) -> Self {
+        let account = account.into();
+        Self {
+            storage_credentials: None,
+            blob_storage_url: get_endpoint_uri(None, &account, "blob").unwrap(),
```

Comment:
I think we'll likely want to rework how we handle urls completely. In Cosmos, we ask the [`CloudLocation` for the right url]((https://github.com/Azure/azure-sdk-for-rust/blob/bc1d085a8d6ba864a7acc0cccbad78efef348561/sdk/data_cosmos/src/clients/cosmos.rs#L132) (more on `CloudLocation` below). 

This should also work here, but we'll just need to include one additional piece of information: the resource type. We can encapsulate that in an enum which gets passed to the `url` method which can then construct the right url based on cloud location and the resource type. 

I think for now, we can just construct the url as a new `String` every time. Once we get an API we're happy with we can add back some sort of memoization should we need to.